### PR TITLE
Fix outputs function

### PR DESCRIPTION
**Pull request by nick-harder**
-rl params were not written when not in learning
-should be for eval
-error handling was wrong

### CODE REVIEW

File: assume/common/units_operator.py [hunk 1]
```diff
@@ -395,8 +395,13 @@ def write_learning_to_output(self, start: datetime, marketconfig: MarketConfig):
...
 
-        db_aid = self.context.data_dict.get("learning_output_agent_id")
-        db_addr = self.context.data_dict.get("learning_output_agent_addr")
+        db_aid = self.context.data_dict.get(
+            "learning_output_agent_id"
+        ) or self.context.data_dict.get("output_agent_id")
+        db_addr = self.context.data_dict.get(
+            "learning_output_agent_addr"
+        ) or self.context.data_dict.get("output_agent_id")
+
```
Comment (maurerle) [inline assume/common/units_operator.py:404]: why is this needed?
it must be output_agent_addr at least..?

File: assume/common/outputs.py [hunk 2]
```diff
@@ -357,7 +357,7 @@ async def on_stop(self):
         for query in queries:
             try:
                 df = pd.read_sql(query, self.db)
-            except (OperationalError, ProgrammingError):
+            except Exception as e:
```
Comment (maurerle) [inline assume/common/outputs.py:360]: We don't want to catch every exception without notice.
You might want to add a second except block with different hadling..?

File: assume/common/units_operator.py (hunk 1 above)
Comment (nick-harder) [reply assume/common/units_operator.py:404]: if it is only output agent, we don't write the RL params at all. It would be still a good idea to write them (for example during evaluations). So this is always written

Comment (nick-harder) [reply assume/common/units_operator.py:404]: oh the address, haha, saw it, thanks

File: assume/common/outputs.py (hunk 2 above)
Comment (nick-harder) [reply assume/common/outputs.py]: yeah true

File: assume/common/units_operator.py (hunk 1 above)
Comment (maurerle) [reply assume/common/units_operator.py:404]: But then you should just add the learning_output_agent config on validation runs too - instead of changing this here?

File: assume/common/outputs.py [hunk 3]
```diff
@@ -359,6 +359,8 @@ async def on_stop(self):
                 df = pd.read_sql(query, self.db)
             except (OperationalError, ProgrammingError):
                 continue
+            except Exception as e:
+                raise e
```
Comment (maurerle) [inline assume/common/outputs.py:363]: This does not do anything - it just catches and raises again.
I think you need this:

```suggestion
            except Exception as e:
                logger.error("could not read query: %s", e)
                continue
```

**Review by maurerle — APPROVED** [review]
